plugins/outputs/loki/stream.go

```diff
@@ -0,0 +1,66 @@
+package loki
+
+import (
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+)
+
+type (
+	Log []string
+
+	Streams struct {
+		Streams []Stream `json:"streams"`
+	}
+
+	Stream struct {
+		key string
+
+		Labels map[string]string `json:"stream"`
+		Logs   []Log             `json:"values"`
+	}
+)
+
+func (s *Streams) insertLog(ts []*telegraf.Tag, l Log) {
+	var (
+		key   = uniqKeyFromTagList(ts)
+		index int
+		found bool
+	)
+
+	for i, s := range s.Streams {
+		if s.key == key {
+			index, found = i, true
+			break
+		}
+	}
```

Comment:
This looks a bit like the poor-man's (slow) version of `map[string][]Stream`. The only reason to do it is the marshaling but I'm not sure it is worth the burden instead of writing a custom marshaller as described in the [json docu](https://golang.org/pkg/encoding/json/#Marshaler)

It might look like this:
```
type Stream struct {
	Labels map[string]string `json:"stream"`
	Logs   []Log             `json:"values"`
}

type Streams map[string][]Stream

func (s Streams) MarshalJSON() ([]byte, error) {
    var out struct {
        S []Stream `json:"streams"`
    }
    for _, streams := range s {
	for _, v := range streams {
	        out.S = append(out.S, v)
	}
    }
    return json.Marshal(out)
}
```

Reply:
I probably can improve this part too, I will do it